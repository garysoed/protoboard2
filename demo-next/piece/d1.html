<style>
  .slotContainer {
    display: inline-block;
    position: relative;
    width: 12rem;
    height: 12rem;
  }
</style>

<pbd-documentation-template mk-body-1 label="D1">
  <div slot="content">
    <p>
      A D1 is the most basic piece. It represents any piece with only one face.
    </p>
    <p>
      Some examples:
    </p>
    <ul>
      <li>
        <p>Meeple</p>
        <div class="slotContainer">
          <pb-slot id="meepleSlot"></pb-slot>
        </div>
      </li>
      <li>
        <p>Gem</p>
        <div class="slotContainer">
          <pb-slot id="gemSlot"></pb-slot>
        </div>
      </li>
    </ul>
  </div>
</pbd-documentation-template>
