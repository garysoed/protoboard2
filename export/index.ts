export {Lens, $lens} from '../src/core/lens';
export {LensDisplay} from '../src/core/lens-display';

// action
export {pickAction} from '../src/action/pick-action';

// coordinate
export {Cartesian} from '../src/coordinate/cartesian';
export {Coordinate} from '../src/coordinate/coordinate';
export {Indexed} from '../src/coordinate/indexed';

// region
export {Deck, $deck, DeckSpec} from '../src/region/deck';
export {Slot, $slot, SlotSpec, slotSpec} from '../src/region/slot';

// core
export {$createSpecEntries, $createSpecMap} from '../src/objects/object-create-spec';
export {ContentSpec} from '../src/payload/is-container';
export {ObjectSpec} from '../src/types/object-spec';
export {$render, Render} from '../src/objects/render';
export {$active, Active, activeSpec} from '../src/core/active';
export {$$activeSpec} from '../src/objects/active-spec';

// piece
export {$d1, D1, D1Spec, d1Spec} from '../src/piece/d1';
export {$d2, D2, D2Spec} from '../src/piece/d2';
export {$d6, D6, D6Spec} from '../src/piece/d6';

// tiling
export {rectOrthogonal, RectOrthogonal, Direction as RectOrthogonalDirection, distance as rectOrthogonalDistance} from '../src/tiling/rect-orthogonal';
export {vhex, VHex, Direction as VHexDirection, distance as vhexDistance} from '../src/tiling/vhex';
export {Tile} from '../src/tiling/types';
