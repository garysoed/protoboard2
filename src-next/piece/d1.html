<div id="container">
  <slot id="slot" name="face-0"></slot>
</div>
