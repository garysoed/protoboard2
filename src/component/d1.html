<style id="theme"></style>
<slot></slot>
