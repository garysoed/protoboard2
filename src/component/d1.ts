import { _p } from '@mask';
import { RotateAction } from 'src/action/rotate-action';

import { PickAction } from '../action/pick-action';
import { BaseComponent } from '../core/base-component';

import template from './d1.html';

@_p.customElement({
  tag: 'pb-d1',
  template,
})
export class D1 extends BaseComponent {
  constructor(shadowRoot: ShadowRoot) {
    super(
        [
          new PickAction(),
          new RotateAction(0, [0, 90, 180, 270]),
        ],
        shadowRoot,
    );
  }
}
