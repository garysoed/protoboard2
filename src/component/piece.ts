import { _p } from '@mask';
import { PickAction } from '../action/pick-action';
import { BaseComponent } from '../core/base-component';
import template from './piece.html';

@_p.customElement({
  tag: 'pb-piece',
  template,
})
export class Piece extends BaseComponent {
  constructor(shadowRoot: ShadowRoot) {
    super(
        [
          new PickAction(),
        ],
        shadowRoot,
    );
  }
}
