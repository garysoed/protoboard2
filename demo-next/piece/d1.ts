import {ICON, renderTheme} from 'mask';
import {Context, Ctrl, id, registerCustomElement} from 'persona';
import {Observable, of} from 'rxjs';

import {D1} from '../../src-next/piece/d1';
import {SLOT} from '../../src-next/region/slot';
import {$state$} from '../demo-state';
import {DOCUMENTATION_TEMPLATE} from '../template/documentation-template';

import template from './d1.html';


const $d1Demo = {
  shadow: {
    gemSlot: id('gemSlot', SLOT, {}),
    meepleSlot: id('meepleSlot', SLOT, {}),
  },
};


class D1Demo implements Ctrl {
  private readonly state$ = $state$.get(this.$.vine)._('d1');

  constructor(private readonly $: Context<typeof $d1Demo>) {
  }

  get runs(): ReadonlyArray<Observable<unknown>> {
    return [
      renderTheme(this.$),
      of(this.state$._('gemSlot')).pipe(this.$.shadow.gemSlot.state()),
      of(this.state$._('meepleSlot')).pipe(this.$.shadow.meepleSlot.state()),
    ];
  }
}

export const D1_DEMO = registerCustomElement({
  ctrl: D1Demo,
  deps: [
    D1,
    DOCUMENTATION_TEMPLATE,
    ICON,
    SLOT,
  ],
  spec: $d1Demo,
  tag: 'pbd-d1',
  template,
});
