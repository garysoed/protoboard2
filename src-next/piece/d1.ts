import {cache} from 'gs-tools/export/data';
import {mapNullableTo} from 'gs-tools/export/rxjs';
import {mutableState} from 'gs-tools/export/state';
import {intersectType, undefinedType, unknownType} from 'gs-types';
import {Context, iattr, icall, query, itarget, ivalue, ocase, ostyle, registerCustomElement, SLOT} from 'persona';
import {Observable} from 'rxjs';

import {pickAction} from '../action/pick-action';
import {DEFAULT_ROTATE_CONFIG, rotateAction, ROTATE_CONFIG_TYPE} from '../action/rotate-action';
import {BaseComponent, create$baseComponent} from '../core/base-component';
import {renderFace} from '../render/render-face';
import {renderRotatable} from '../render/render-rotatable';
import {ComponentState} from '../types/component-state';
import {IsRotatable} from '../types/is-rotatable';
import {TriggerType, TRIGGER_SPEC_TYPE} from '../types/trigger-spec';

import template from './d1.html';


export interface D1State extends ComponentState, IsRotatable {
  readonly face: unknown;
}

/**
 * The D1's API.
 *
 * @thModule piece
 */
const $d1 = {
  host: {
    ...create$baseComponent<D1State>().host,
    height: iattr('height'),
    pick: icall('pick', undefinedType),
    pickConfig: ivalue('pickConfig', TRIGGER_SPEC_TYPE, {type: TriggerType.CLICK}),
    rotate: icall('rotate', undefinedType),
    rotateConfig: ivalue(
        'rotateConfig',
        intersectType([TRIGGER_SPEC_TYPE, ROTATE_CONFIG_TYPE]),
        {...DEFAULT_ROTATE_CONFIG, type: TriggerType.R},
    ),
    width: iattr('width'),
  },
  shadow: {
    container: query('#container', SLOT, {
      height: ostyle('height'),
      face: ocase(unknownType),
      target: itarget(),
      transform: ostyle('transform'),
      width: ostyle('width'),
    }),
  },
};

export function d1State(id: {}, face: unknown, partial: Partial<D1State> = {}): D1State {
  return {
    id,
    face,
    rotationDeg: mutableState(0),
    ...partial,
  };
}

class D1Ctrl extends BaseComponent<D1State> {
  /**
   * @internal
   */
  constructor(private readonly $: Context<typeof $d1>) {
    super($, 'D1');
  }

  @cache()
  get runs(): ReadonlyArray<Observable<unknown>> {
    return [
      ...super.runs,
      this.state._('face').pipe(
          this.$.shadow.container.face(faceId => renderFace(this.$.vine, faceId)),
      ),
      this.$.host.height.pipe(mapNullableTo(''), this.$.shadow.container.height()),
      this.$.host.width.pipe(mapNullableTo(''), this.$.shadow.container.width()),
      this.installAction(
          pickAction,
          'Pick',
          this.$.shadow.container.target,
          this.$.host.pickConfig,
          this.$.host.pick,
      ),
      this.installAction(
          rotateAction,
          'Rotate',
          this.$.shadow.container.target,
          this.$.host.rotateConfig,
          this.$.host.rotate,
      ),
      this.state.$('rotationDeg').pipe(
          renderRotatable(),
          this.$.shadow.container.transform(),
      ),
    ];
  }
}

export const D1 = registerCustomElement({
  ctrl: D1Ctrl,
  spec: $d1,
  tag: 'pb-d1',
  template,
});