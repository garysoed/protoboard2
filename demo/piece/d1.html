<style>
  .slotContainer {
    display: inline-block;
    position: relative;
    width: 12rem;
    height: 12rem;
  }
</style>

<pbd-documentation-template mk-body-1 label="D1">
  <div slot="content">
    <p>
      A D1 is the most basic piece. It represents any piece with only one face.
    </p>
    <p>
      Some examples:
    </p>
    <ul>
      <li>
        <p>Meeple</p>
        <div class="slotContainer">
          <pb-surface id="meepleSlot"></pb-surface>
        </div>
      </li>
      <li>
        <p>Gem</p>
        <div class="slotContainer">
          <pb-surface id="gemSlot"></pb-surface>
        </div>
      </li>
    </ul>
  </div>
</pbd-documentation-template>
