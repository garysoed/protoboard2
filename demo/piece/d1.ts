import { cache } from 'gs-tools/export/data';
import { debug, filterNonNull, mapNonNull } from 'gs-tools/export/rxjs';
import { elementWithTagType } from 'gs-types';
import { $icon, _p, Icon, registerSvg, ThemedCustomElementCtrl } from 'mask';
import { element, onDom, PersonaContext, renderCustomElement } from 'persona';
import { BehaviorSubject, Observable, of as observableOf } from 'rxjs';
import { map, switchMap, tap, withLatestFrom } from 'rxjs/operators';

import { $d1 as $d1Impl, D1 as D1Impl } from '../../src/piece/d1';
import { registerStateHandler } from '../../src/state/state-service';
import coinSvg from '../asset/coin.svg';
import gemSvg from '../asset/gem.svg';
import meepleSvg from '../asset/meeple.svg';
import { $stagingService } from '../core/staging-service';
import { $pieceTemplate, PieceTemplate } from '../template/piece-template';

import template from './d1.html';


export const $d1 = {
  tag: 'pbd-d1',
  api: {},
};

const $ = {
  create: element('create', elementWithTagType('section'), {}),
  customize: element('customize', elementWithTagType('section'), {
    onClick: onDom('click'),
  }),
  previewIcon: element('previewIcon', $icon, {}),
  template: element('template', $pieceTemplate, {}),
};

const D1_PREVIEW_TYPE = 'preview-d1';

interface D1PreviewPayload {
  readonly icon: string;
}

@_p.customElement({
  ...$d1,
  configure: vine => {
    registerSvg(vine, 'meeple', {type: 'embed', content: meepleSvg});
    registerSvg(vine, 'coin', {type: 'embed', content: coinSvg});
    registerSvg(vine, 'gem', {type: 'embed', content: gemSvg});

    registerStateHandler<D1PreviewPayload>(
        D1_PREVIEW_TYPE,
        (state, context) => {
          const icon$ = renderCustomElement(
              $icon,
              {inputs: {icon: state.payload.icon}},
              context,
          );
          return renderCustomElement(
              $d1Impl,
              {
                children: icon$.pipe(map(el => [el])),
                inputs: {objectId: observableOf(state.id)},
              },
              context,
          );
        },
        vine,
    );
  },
  dependencies: [
    PieceTemplate,
    Icon,
    D1Impl,
  ],
  template,
})
export class D1 extends ThemedCustomElementCtrl {
  private readonly selectedIcon$ = new BehaviorSubject<string>('meeple');

  constructor(context: PersonaContext) {
    super(context);
    this.addSetup(this.handleOnCustomizeClick$);
    this.render($.previewIcon._.icon, this.selectedIcon$);
    this.addSetup(this.handleOnPieceAdd$);
  }

  @cache()
  private get handleOnCustomizeClick$(): Observable<unknown> {
    return this.declareInput($.customize._.onClick)
        .pipe(
            map(event => event.target),
            mapNonNull(target => {
              if (!(target instanceof Element)) {
                return null;
              }

              if (!target.classList.contains('bigButton')) {
                return null;
              }

              return target.id;
            }),
            filterNonNull(),
            tap(icon => this.selectedIcon$.next(icon)),
        );
  }

  @cache()
  private get handleOnPieceAdd$(): Observable<unknown> {
    return this.declareInput($.template._.onAdd).pipe(
        withLatestFrom(this.selectedIcon$, $stagingService.get(this.vine)),
        switchMap(([, icon, stagingService]) => {
          return stagingService.addState(D1_PREVIEW_TYPE, {icon});
        }),
    );
  }
}
