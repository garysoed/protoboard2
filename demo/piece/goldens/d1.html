<pbd-d1><style>
  .slotContainer {
    display: inline-block;
    position: relative;
    width: 12rem;
    height: 12rem;
  }
</style>

<pbd-documentation-template mk-body-1="" label="D1"><style>
  #content::slotted(*) {
    padding-left: 1rem;
    padding-right: 1rem;
  }

  #title {
    padding-left: 1rem;
    padding-right: 1rem;
  }
</style>

<section flex="">
  <h3 id="title" mk-display="">D1</h3>
  <slot id="content" name="content"><div slot="content">
    <p>
      A D1 is the most basic piece. It represents any piece with only one face.
    </p>
    <p>
      Some examples:
    </p>
    <ul>
      <li>
        <p>Meeple</p>
        <div class="slotContainer">
          <pb-slot id="meepleSlot" object-id="6"><style>
  #root {
    align-items: center;
    justify-content: center;
    background-color: var(--mkThemeBG);
    border: 1px solid var(--mkThemeFG);
    box-sizing: border-box;
    flex-wrap: wrap;
  }
</style>

<div id="root" fill-parent="" mk-action-2="" layout="row">
  <!-- #content--><pb-d1 object-id="2"><slot id="slot" name="face-0"><pbd-rendered-face face-type="meeple" slot="face-0"><style>
  #summary {
    height: 4rem;
    width: 4rem;
  }
</style>
<pb-lens><style>
  #details {
    display: none;
  }
</style>

<div id="root">
  <slot name="summary"><div id="summary" slot="summary">
    <mk-icon id="icon" aria-hidden="true" role="presentation" icon="meeple"><style>
  :host {
    pointer-events: none;
    user-select: none;
  }

  #root > svg #foreground {
    fill: var(--mkThemeFG) !important;
  }
</style>
<span id="root">
  <!-- #content --><svg xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:cc="http://creativecommons.org/ns#" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns:svg="http://www.w3.org/2000/svg" xmlns="http://www.w3.org/2000/svg" xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd" xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape" sodipodi:docname="meeple.svg" inkscape:version="1.0 (4035a4f, 2020-05-01)" id="svg4170" xml:space="preserve" enable-background="new 0 0 100 100" viewBox="0 0 24 24" y="0px" x="0px" version="1.1" fill="#000000" height="100%"><metadata id="metadata4178"><rdf:RDF><cc:Work rdf:about=""><dc:format>image/svg+xml</dc:format><dc:type rdf:resource="http://purl.org/dc/dcmitype/StillImage"></dc:type><dc:title></dc:title></cc:Work></rdf:RDF></metadata><defs id="defs4176"></defs><sodipodi:namedview inkscape:document-rotation="0" inkscape:current-layer="svg4170" inkscape:window-maximized="0" inkscape:window-y="23" inkscape:window-x="0" inkscape:cy="50" inkscape:cx="26.655897" inkscape:zoom="6.19" showgrid="false" id="namedview4174" inkscape:window-height="825" inkscape:window-width="1393" inkscape:pageshadow="2" inkscape:pageopacity="0" guidetolerance="10" gridtolerance="10" objecttolerance="10" borderopacity="1" bordercolor="#666666" pagecolor="#ffffff"></sodipodi:namedview><path style="clip-rule:evenodd;fill-rule:evenodd;stroke-width:0.24" inkscape:connector-curvature="0" id="foreground" d="M 23.998076,10.324561 C 24.136796,7.0228813 16.233117,5.4153614 15.890637,5.0390414 15.719997,4.8518415 16.286637,-0.00359817 11.999997,1.8323824e-6 7.7133575,-0.00359817 8.2797575,4.8518415 8.1091175,5.0390414 c -0.34248,0.37632 -8.2456793,1.9838399 -8.10719931,5.2855196 0.13823999,3.30192 4.33223961,1.60608 5.22671951,2.57784 0.7824,0.85056 -3.6669596,6.294959 -4.1647196,9.752159 C 0.90167812,23.78136 1.2184781,24 2.313598,24 c 1.9862398,0 3.8824797,-0.0012 5.6287195,-0.0012 0.78888,0 1.0694399,-0.45336 1.4764799,-1.07808 0.9009606,-1.38288 1.9891206,-3.60576 2.5811996,-3.60552 0.59184,-2.4e-4 1.68024,2.22264 2.5812,3.60528 0.40704,0.62496 0.68736,1.07808 1.47624,1.07808 1.746,0 3.64248,0.0012 5.62872,0.0012 1.09512,0 1.41216,-0.21864 1.24968,-1.34544 -0.49752,-3.4572 -4.94664,-8.901599 -4.16472,-9.752159 0.89448,-0.97152 5.088959,0.72432 5.226959,-2.5776 z"></path></svg>
</span>
<link rel="stylesheet" href="base/test.css"></mk-icon>
  </div></slot>
</div>
<slot id="details" name="details"><div slot="details">
    <h3 id="name">Meeple</h3>
    <p mk-body-1="">More detailed information on the piece goes here.</p>
  </div></slot><link rel="stylesheet" href="base/test.css"></pb-lens><link rel="stylesheet" href="base/test.css"></pbd-rendered-face></slot>
<link rel="stylesheet" href="base/test.css"></pb-d1>
</div>
<link rel="stylesheet" href="base/test.css"></pb-slot>
        </div>
      </li>
      <li>
        <p>Gem</p>
        <div class="slotContainer">
          <pb-slot id="gemSlot" object-id="8"><style>
  #root {
    align-items: center;
    justify-content: center;
    background-color: var(--mkThemeBG);
    border: 1px solid var(--mkThemeFG);
    box-sizing: border-box;
    flex-wrap: wrap;
  }
</style>

<div id="root" fill-parent="" mk-action-2="" layout="row">
  <!-- #content--><pb-d1 object-id="4"><slot id="slot" name="face-0"><pbd-rendered-face face-type="gem" slot="face-0"><style>
  #summary {
    height: 4rem;
    width: 4rem;
  }
</style>
<pb-lens><style>
  #details {
    display: none;
  }
</style>

<div id="root">
  <slot name="summary"><div id="summary" slot="summary">
    <mk-icon id="icon" aria-hidden="true" role="presentation" icon="gem"><style>
  :host {
    pointer-events: none;
    user-select: none;
  }

  #root > svg #foreground {
    fill: var(--mkThemeFG) !important;
  }
</style>
<span id="root">
  <!-- #content --><svg xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:cc="http://creativecommons.org/ns#" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns:svg="http://www.w3.org/2000/svg" xmlns="http://www.w3.org/2000/svg" xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd" xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape" sodipodi:docname="gem.svg" inkscape:version="1.0 (4035a4f, 2020-05-01)" xml:space="preserve" viewBox="0 0 24 24" height="100%" y="0px" x="0px" id="Capa_1" version="1.1"><metadata id="metadata4240"><rdf:RDF><cc:Work rdf:about=""><dc:format>image/svg+xml</dc:format><dc:type rdf:resource="http://purl.org/dc/dcmitype/StillImage"></dc:type><dc:title></dc:title></cc:Work></rdf:RDF></metadata><defs id="defs4238"></defs><sodipodi:namedview inkscape:document-rotation="0" inkscape:current-layer="Capa_1" inkscape:window-maximized="0" inkscape:window-y="23" inkscape:window-x="0" inkscape:cy="58.978027" inkscape:cx="-43.197886" inkscape:zoom="2.8405145" showgrid="false" id="namedview4236" inkscape:window-height="792" inkscape:window-width="1323" inkscape:pageshadow="2" inkscape:pageopacity="0" guidetolerance="10" gridtolerance="10" objecttolerance="10" borderopacity="1" bordercolor="#666666" pagecolor="#ffffff"></sodipodi:namedview><path style="stroke-width:0.0392157" d="m 10.413765,2.9165315 0.500862,-0.582785 H 5.4396078 L 0,8.6057075 H 5.5206667 Z M 6.8312941,8.6061385 H 17.171686 L 14.360549,5.3364135 12.002118,2.5935505 8.438,6.7374725 Z M 0.06615686,9.5997465 10.346392,20.417551 5.4302353,9.5997465 Z M 17.483098,9.5989235 h -0.895922 l -3.766784,8.23e-4 H 6.5198431 l 5.4494119,11.9901955 0.03502,0.07635 5.192117,-11.434274 z M 13.66902,20.402179 23.93302,9.6001785 H 18.573608 Z M 24,8.6057075 l -5.441333,-6.272 H 13.08749 l 5.39353,6.272 z" id="foreground" inkscape:connector-curvature="0"></path><g id="g4206"></g><g id="g4208"></g><g id="g4210"></g><g id="g4212"></g><g id="g4214"></g><g id="g4216"></g><g id="g4218"></g><g id="g4220"></g><g id="g4222"></g><g id="g4224"></g><g id="g4226"></g><g id="g4228"></g><g id="g4230"></g><g id="g4232"></g><g id="g4234"></g></svg>
</span>
<link rel="stylesheet" href="base/test.css"></mk-icon>
  </div></slot>
</div>
<slot id="details" name="details"><div slot="details">
    <h3 id="name">Gem</h3>
    <p mk-body-1="">More detailed information on the piece goes here.</p>
  </div></slot><link rel="stylesheet" href="base/test.css"></pb-lens><link rel="stylesheet" href="base/test.css"></pbd-rendered-face></slot>
<link rel="stylesheet" href="base/test.css"></pb-d1>
</div>
<link rel="stylesheet" href="base/test.css"></pb-slot>
        </div>
      </li>
    </ul>
  </div></slot>
</section>
<link rel="stylesheet" href="base/test.css"></pbd-documentation-template>
<link rel="stylesheet" href="base/test.css"></pbd-d1>